lib/ace/ext/papeeria/katex-previewer.coffee

```diff
@@ -53,49 +244,60 @@ define((require, exports, module) ->
       { type: "rparen", value: "}" }
     ]
 
+    # empty constructor
     constructor: (@editor) ->
 
-    equalTokens: (token1, token2) ->
-      if token1? and token2?
-        return token1.type == token2.type and token1.value == token2.value
-      else return if token1? or token2? then false else true
+    @equalTokens: (token1, token2) ->
+      return token1.type == token2.type and token1.value == token2.value
 
     getEquationStart: (tokenIterator) ->
+      # if tokenIterator initialy is on the empty line, it's current token is null
+      if not tokenIterator.getCurrentToken()?
```

Comment:
Sorry? What do I see here in the comment: `getCurrentToken() == null` when tokeniterator is on the empty line. Is it true that every time when it reaches empty line during iteration its current token becomes null?